app/src/main/java/io/mercury/coroutinesandbox/view/downloader/DownloadListFeature.kt

```diff
@@ -0,0 +1,47 @@
+package io.mercury.coroutinesandbox.view.downloader
+
+import io.mercury.coroutinesandbox.interactors.GetDownloadedMovies
+import io.mercury.coroutinesandbox.interactors.GetInProgressDownloads
+import io.mercury.coroutinesandbox.interactors.GetInProgressDownloads.InProgressDownload
+import io.mercury.coroutinesandbox.view.downloader.DownloadListFeature.State.Loaded
+import io.mercury.coroutinesandbox.view.downloader.DownloadListFeature.State.Uninitialized
+import io.mercury.domain.models.Movie
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.asStateFlow
+import kotlinx.coroutines.flow.collect
+import kotlinx.coroutines.flow.combine
+import kotlinx.coroutines.launch
+
+class DownloadListFeature(
+    private val scope: CoroutineScope,
+    private val getInProgressDownloads: GetInProgressDownloads,
+    private val getDownloadedMovies: GetDownloadedMovies
+) {
+
+    private val statePublisher = MutableStateFlow<State>(Uninitialized)
+    val state get() = statePublisher.asStateFlow()
+
+    fun load() {
+        if (state.value is Uninitialized) {
+            scope.launch(Dispatchers.IO) {
```

Comment:
For testability we should consider providing the dispatcher. 

However, mockk can mock objects and Dispatchers is an object, so perhaps this is fine?

Reply:
Makes sense to pass it in, but it does make the constructor a little heavier. I'd have to look at how you mock the dispatchers, but I like the idea of explicitly injecting dispatchers and scope... I think.